File: JavaScript/01.JavaScript-Basic-February-2021/10.whileLoop-Exersice/03.vacation.js
function vacation(input) {
    let index = 0;

    let needMoney = Number(input[index]);
    index++;

    let money = Number(input[index]);
    index++;

    let counter = 0;
    let dayCounter = 0;
    let isSaveMoney = true;
    while (money < needMoney) {
        dayCounter++;
        let action = input[index];
        index++;

        let tempMoney = Number(input[index]);
        index++;

        if (action === "save") {
            money += tempMoney;
            counter = 0;
        } else {
            money -= tempMoney;
            if (money < 0) {
                money = 0;
            }
            counter++;
        }

        if (counter === 5) {
            console.log(`You can't save the money.`);
            console.log(dayCounter);
            isSaveMoney = false;
            break;
        }
    }

    if (isSaveMoney) {
        console.log(`You saved the money for ${dayCounter} days.`);
    }
}

vacation(["2000", "1000", "spend", "1200", "save", "2000"]);
vacation(["110", "60", "spend", "10", "spend", "10", "spend", "10", "spend", "10", "spend", "10"]);

File: Python/01.Python-Basic-April-2019/09.while_loop_exercise/03.vacation.py
money_needed = float(input())
start_money = float(input())

total_days_counter = 0
spend_days_counter = 0

while True:
    action = input()
    money = float(input())
    total_days_counter += 1

    if action == 'spend':
        start_money -= money
        if start_money < 0:
            start_money = 0

        spend_days_counter += 1

        if spend_days_counter == 5:
            print(f"You can't save the money.\n{total_days_counter}")
            break

    elif action == 'save':
        spend_days_counter = 0

        start_money += money

        if start_money >= money_needed:
            print(f"You saved the money for {total_days_counter} days.")
            break
